staging/src/k8s.io/apiserver/pkg/cel/library/cost.go

```diff
@@ -126,13 +126,32 @@ func (l *CostEstimator) EstimateCallCost(function, overloadId string, target *ch
 			sz := l.sizeEstimate(*target)
 			toReplaceSz := l.sizeEstimate(args[0])
 			replaceWithSz := l.sizeEstimate(args[1])
-			// smallest possible result: smallest input size composed of the largest possible substrings being replaced by smallest possible replacement
-			minSz := uint64(math.Ceil(float64(sz.Min)/float64(toReplaceSz.Max))) * replaceWithSz.Min
-			// largest possible result: largest input size composed of the smallest possible substrings being replaced by largest possible replacement
-			maxSz := uint64(math.Ceil(float64(sz.Max)/float64(toReplaceSz.Min))) * replaceWithSz.Max
+
+			var perReplaceWithSz, retainedSz checker.SizeEstimate
```

Comment:
this name is unclear - is `perReplaceWithSz` the min/max of a single replace, or the min/max number of times we can replace? if the latter, maybe `replaceCount` would be clearer

Reply:
Thx, I clearly was working too late last night :)  Fixed!